src/fdwave_raja/target_3d.cpp

```diff
@@ -0,0 +1,300 @@
+                                   +vpViewConst(i,j,k)*(lap+phiView(i,j,k)) )
+                                 / (1.f+2.f*etaViewConst(i+1,j+1,k+1));
+                               phiView(i,j,k) =
+                                 ( phiView(i,j,k)-
+                                   ( (etaViewConst(i+2,j+1,k+1)-etaViewConst(i,j+1,k+1))
+                                     *(uViewConst(i+1+lx,j+ly,k+lz)-uViewConst(i-1+lx,j+ly,k+lz))*hdx_2
+                                     +(etaViewConst(i+1,j+2,k+1)-etaViewConst(i+1,j,k+1))
+                                     *(uViewConst(i+lx,j+1+ly,k+lz)-uViewConst(i+lx,j-1+ly,k+lz))*hdy_2
+                                     +(etaViewConst(i+1,j+1,k+2)-etaViewConst(i+1,j+1,k))
+                                     *(uViewConst(i+lx,j+ly,k+1+lz)-uViewConst(i+lx,j+ly,k-1+lz))*hdz_2 ) )
+                                 / (1.f+etaViewConst(i+1,j+1,k+1));
+                             });
+}
+
+void target_pml_3d(llint nx, llint ny,llint nz,
+                   llint x3, llint x4, llint y3,
+                   llint y4, llint z3, llint z4,
+                   llint lx, llint ly, llint lz,
+                   float hdx_2, float hdy_2, float hdz_2,
+                   const float *__restrict__ coefx,
+                   const float *__restrict__ coefy,
+                   const float *__restrict__ coefz,
+                   const float *__restrict__ u,
+                   float *__restrict__ v,
+                   const float *__restrict__ vp,
+                   float *__restrict__ phi,
+                   const float *__restrict__ eta)
+{
+  RAJA::View< const float, RAJA::Layout<1, RAJA::Index_type, 0> > coefxViewConst( coefx, 5 );
+  RAJA::View< const float, RAJA::Layout<1, RAJA::Index_type, 0> > coefyViewConst( coefy, 5 );
+  RAJA::View< const float, RAJA::Layout<1, RAJA::Index_type, 0> > coefzViewConst( coefz, 5 );
+
+  RAJA::View< const float, RAJA::Layout<3, RAJA::Index_type, 2> > uViewConst( u, RAJA::Layout<3>(nx+2*lx, ny+2*ly, nz+2*lz) );
+  RAJA::View< float, RAJA::Layout<3, RAJA::Index_type, 2> > vView( v, RAJA::Layout<3>(nx+2*lx, ny+2*ly, nz+2*lz) );
+  RAJA::View< const float, RAJA::Layout<3, RAJA::Index_type, 2> > vpViewConst( vp, RAJA::Layout<3>(nx, ny, nz) );
+  RAJA::View< float, RAJA::Layout<3, RAJA::Index_type, 2> > phiView( phi, RAJA::Layout<3>(nx, ny, nz) );
+  RAJA::View< const float, RAJA::Layout<3, RAJA::Index_type, 2> > etaViewConst( eta, RAJA::Layout<3>(nx+2, ny+2, nz+2) );    
+
+  compute_pml_3d( RAJA::Index_type(x3), RAJA::Index_type(x4),
+                  RAJA::Index_type(y3), RAJA::Index_type(y4),
+                  RAJA::Index_type(z3), RAJA::Index_type(z4),
+                  RAJA::Index_type(lx), RAJA::Index_type(ly), RAJA::Index_type(lz),
+                  hdx_2, hdy_2, hdz_2,
+                  coefxViewConst, coefyViewConst, coefzViewConst,
+                  uViewConst, vpViewConst, etaViewConst, vView, phiView );
```

Comment:
@corbett5 This is the main kernel of the code. To implement it using RAJA, we followed the examples with FD stencils in the RAJA performance suite. 

Here is the problem that we have: if you uncomment the code above (`compute_pml_3d`), and instead use the standard `for` loops below, the code still produces the same results, but is twice faster on the CPU, and we don't understand why. We must be doing something wrong, but I could not find what. If you spot something that we are doing wrong in `compute_pml_3d`, please let us know.